test/functional/minchainwork.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+"""Test logic for setting nMinimumChainWork on command line.
+
+Nodes don't consider themselves out of "initial block download" until
+their active chain has more work than nMinimumChainWork.
+
+While in initial block download, nodes won't relay blocks to their peers, so
+test that this parameter functions as intended by verifying that block relay
+only succeeds past a given node once its nMinimumChainWork has been exceeded.
+"""
+
+import time
+
+from test_framework.test_framework import BitcoinTestFramework
+from test_framework.util import sync_blocks, connect_nodes
+
+# 2 hashes required per regtest block (with no difficulty adjustment)
+REGTEST_WORK_PER_BLOCK = 2
+
+class MinimumChainWorkTest(BitcoinTestFramework):
+    def __init__(self):
+        super().__init__()
+        self.setup_clean_chain = True
+        self.num_nodes = 4
+        self.extra_args = [[], ["-minimumchainwork=0x65"], ["-minimumchainwork=0xff"], ["-minimumchainwork=0x01ff"]]
+        self.node_min_work = [0, 101, 255, 511]
+
+    def setup_network(self):
+        # This test relies on the chain setup being:
+        # node0 <- node1 <- node2 <- node3
+        # Before leaving IBD, nodes prefer to download blocks from outbound
+        # peers, so ensure that we're mining on an outbound peer and testing
+        # block relay to inbound peers.
+        self.setup_nodes()
+        for i in range(self.num_nodes-1):
+            connect_nodes(self.nodes[i+1], i)
+
+    def run_test(self):
+        # Start building a chain on node0.  node2 shouldn't be able to sync until node1's
+        # minchainwork is exceeded; node3 shouldn't be able to sync until node2's minchainwork
+        # is exceeded.
+        cumulative_chain_work = 2 # Genesis block's work
+        for i in [1,2]:
+            self.log.info("Testing relay across node %d (minChainWork = %d)", i, self.node_min_work[i])
+
+            starting_blockcount = self.nodes[i+1].getblockcount()
+
+            num_blocks_to_generate = int((self.node_min_work[i] - cumulative_chain_work) / 2)
```

Comment:
I think this `2` refers to work per block? In that case, replace with `REGTEST_WORK_PER_BLOCK`